airbyte-integrations/connectors/source-mongodb/lib/mongodb_state.rb

```diff
@@ -0,0 +1,32 @@
+require_relative './airbyte_protocol.rb'
+require_relative './airbyte_logger.rb'
+
+require 'json'
+
+class MongodbState
+
+  def initialize(state_file:)
+    @state = if state_file
+               JSON.parse(File.read(state_file))
+             else
+               {}
+             end
+  end
+
+  def get(stream_name:, cursor_field:)
+    @state.dig(stream_name, cursor_field)
+  end
+
+  def set(stream_name:, cursor_field:, cursor:)
+    @state[stream_name] ||= {}
+    @state[stream_name][cursor_field] = cursor
+  end
+
+  def dump_state!
+    json = @state.to_json
+
+    AirbyteLogger.log("Saving state:\n#{JSON.pretty_generate(@state)}")
+
+    File.write(state_file, @state.to_json)
```

Comment:
should this be logging state to stdout as a state message?

Reply:
I've missed that somehow. Fixed now